ui/src/components/Tasks/AddSample.js

```diff
@@ -1,138 +1,138 @@
-    this.handleCancel = this.handleCancel.bind(this);
-    this.getDefaultSampleData = this.getDefaultSampleData.bind(this);
-    this.handleKeyPress = this.handleKeyPress.bind(this);
-  }
+function getSampleData(params) {
+  return {
+    ...params,
+    type: 'Sample',
+    defaultPrefix: `${params.proteinAcronym}-${params.sampleName}`,
+    location: 'Manual',
+    loadable: true,
+    tasks: [],
+  };
+}
 
-  getDefaultSampleData(params) {
-    let prefix = params.sampleName ? params.sampleName : 'noname';
+function AddSample(props) {
+  const {
+    show,
+    hide,
+    addSamplesToList,
+    addSampleAndMount,
+    addSamplesToQueue,
+    showList,
+  } = props;
 
-    if (params.proteinAcronym && params.sampleName) {
-      prefix = `${params.proteinAcronym}-${prefix}`;
-    }
+  const { register, formState, handleSubmit, setFocus } = useForm();
+  const { isSubmitted, errors } = formState;
 
-    return {
-      ...params,
-      type: 'Sample',
-      defaultPrefix: prefix,
-      location: 'Manual',
-      loadable: true,
-      tasks: [],
-    };
-  }
+  const { pathname } = useLocation();
 
-  handleCancel() {
-    this.props.hide();
-  }
-
-  _addAndEnqueue(params) {
-    const sampleData = this.getDefaultSampleData(params);
-    this.props.addToQueue(sampleData);
-    this.props.hide();
-  }
-
-  addAndEnqueue() {
-    this.props.handleSubmit(this._addAndEnqueue)();
-  }
+  useEffect(() => {
+    if (show) {
+      // Timeout required when creating new sample from "Samples" page
+      setTimeout(() => setFocus('sampleName'), 0);
+    }
+  }, [setFocus, show]);
 
-  _addAndMount(params) {
-    const sampleData = this.getDefaultSampleData(params);
-    this.props.addAndMount(sampleData);
-    this.props.hide();
-  }
+  function addAndMount(params) {
+    const sampleData = getSampleData(params);
+    addSamplesToList([sampleData]);
+    addSampleAndMount(sampleData);
+    hide();
 
-  addAndMount() {
-    this.props.handleSubmit(this._addAndMount)();
+    if (pathname === '/' || pathname === '/datacollection') {
+      // Switch to mounted sample tab
+      showList('current');
+    }
   }
 
-  handleKeyPress(target) {
-    if (target.charCode === 13) {
-      this.addAndMount();
-    }
+  function addAndQueue(params) {
+    const sampleData = getSampleData(params);
+    addSamplesToList([sampleData]);
+    addSamplesToQueue([sampleData]);
+    hide();
   }
 
-  render() {
-    return (
-      <Modal
-        // fullscreen
-        show={this.props.show}
-        onHide={this.handleCancel}
-        // style={{ width: '60%', height: '20%'}}
-      >
+  return (
+    <Modal show={show} onHide={hide} data-default-styles>
+      <Form noValidate onSubmit={handleSubmit(addAndMount)}>
         <Modal.Header closeButton>
-          <Modal.Title>Add Sample Manually</Modal.Title>
+          <Modal.Title>New Sample</Modal.Title>
         </Modal.Header>
         <Modal.Body>
-          <Form>
-            <FieldsRow>
```

Comment:
I've removed the abstractions that were hiding away the Bootstrap layout and form fields rendering logic. Perhaps I will re-abstract things a bit later on, but for now, there's no need to do so.